web-console/e2e-tests/auto-compaction.spec.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import axios from 'axios';
+import { execSync } from 'child_process';
+import path from 'path';
+import * as playwright from 'playwright-core';
+import { v4 as uuid } from 'uuid';
+
+import { CompactionConfig } from './component/datasources/compaction';
+import { CompactionHashPartitionsSpec } from './component/datasources/compaction';
+import { Datasource } from './component/datasources/datasource';
+import { DatasourcesOverview } from './component/datasources/overview';
+import { saveScreenshotIfError } from './util/debug';
+import { COORDINATOR_URL } from './util/druid';
+import { DRUID_DIR } from './util/druid';
+import { UNIFIED_CONSOLE_URL } from './util/druid';
+import { createBrowserNormal as createBrowser } from './util/playwright';
```

Comment:
For reviewers: To disable headless mode, change `createBrowserNormal` to `createBrowserDebug`

Reply:
I think in the future we could make this be driven by an environment variable